src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -154,6 +154,73 @@ public void addLoan(Loan loan) {
         loanRecords.addLoan(loan);
     }
 
+    /**
+     * Replaces an existing {@code Loan} object in LoanRecords with an edited one.
+     *
+     * @param existingLoan Existing {@code Loan} object to be replaced.
+     * @param updatedLoan Updated {@code Loan} object.
+     */
+    @Override
+    public void updateLoan(Loan existingLoan, Loan updatedLoan) {
+        requireAllNonNull(existingLoan, updatedLoan);
+        loanRecords.updateLoan(existingLoan, updatedLoan);
+    }
+
+    /**
+     * Pays the outstanding fines of the servingBorrower.
+     *
+     * @param amountInCents Amount borrower is paying in cents.
+     * @return Leftover amount in cents, i.e., change to be given.
+     */
+    @Override
+    public int payFine(int amountInCents) {
+        Borrower serving = getServingBorrower();
+        LoanList updatedReturnedLoanList = new LoanList();
+
+        int change = payLoansFine(serving.getReturnedLoanList(), updatedReturnedLoanList, amountInCents);
+
+        Borrower updatedBorrower = new Borrower(serving.getName(), serving.getPhone(), serving.getEmail(),
+                serving.getBorrowerId(), serving.getCurrentLoanList(), updatedReturnedLoanList);
+        borrowerRecords.setBorrower(serving, updatedBorrower);
+
+        setServingBorrower(updatedBorrower);
+
+        return change;
+    }
+
+    /**
+     * Private helper method to help iterate through each Loan in the returnedLoanList and then to update each Loan
+     * object with the maximum possible amount paid.
+     * Returns the leftover amount, i.e., the change.
+     */
+    private int payLoansFine(LoanList origReturnedLoanList, LoanList updatedReturnedLoanList, int amountInCents) {
```

Comment:
Feels like a huge chunk of code with nested branches here, not sure if we can apply SLAP to make it easier to read